### ISSUE TITLE
PHP 7.2 has backward incompatible changes (BC) which prevents the editor from adding new contacts

### ISSUE THREAD

**Issue by higidi** [sulu/sulu#3925]
| Q | A
| --- | ---
| Bug? | yes
| New Feature? | no
| Sulu Version | 1.6.17
| Browser Version | Google Chrome Version 66.0.3359.11

#### Actual Behavior

Throws an error when saving a new contact via Admin backend

#### Expected Behavior

Create the new contact

#### Steps to Reproduce

- Ensure you will run Sulu with PHP 7.2
- Follow the getting started instructions to setup a new Sulu 1.6.17 project
- Login to the admin panel
- Add new contact (without bankaccount!!!)

See: http://php.net/manual/de/migration72.incompatible.php#migration72.incompatible.warn-on-non-countable-types

**vendor/sulu/sulu/src/Sulu/Bundle/ContactBundle/Contact/AbstractContactManager.php** line 1796
```php
    /**
     * TODO: this is just a hack to avoid relations that start with index != 0
     * otherwise deserialization process will parse relations as object instead of an array
     * reindex entities.
     *
     * @param mixed $entities
     */
    private function resetIndexOfSubentites($entities)
    {
        if (count($entities) > 0 && method_exists($entities, 'getValues')) {
            $newEntities = $entities->getValues();
            $entities->clear();
            foreach ($newEntities as $value) {
                $entities->add($value);
            }
        }

        return $entities;
    }
```

![image](https://user-images.githubusercontent.com/474738/39208384-1d53fa82-4803-11e8-90d9-a9122f40b671.png)

**Comment by trickreich** [comment]
@sulu/core-team Sounds like a bug for me?

**Comment by danrot** [comment]
It definitely is... I think we should really get the entire build process running in PHP 7.2, but it's not such an easy task... IIRC we would have to upgrade to a newer PHPUnit version, which would result in quite some work...

@higidi Since you've already found the exact error anyway, could you imagine making a PR? IIRC that was usually just setting a default value on the class member.

**Comment by higidi** [comment]
@danrot yes, i can :-)

**Comment by higidi** [comment]
@danrot done :-)

**Comment by danrot** [comment]
Fixed in #3925, thanks again for your effort!
